lib/pipelinemodel.js

```diff
@@ -0,0 +1,134 @@
+'use strict';
+const async = require('async');
+const hashr = require('screwdriver-hashr');
+const hoek = require('hoek');
+const table = 'pipelines';
+
+const PipelineModel = class {
+    /**
+     * Construct a PipelineModel object
+     * @method constructor
+     * @param  {Object}    datastore         Object that will perform operations on the datastore
+     */
+    constructor(datastore) {
+        this.datastore = datastore;
+    }
+
+    /**
+     * Create a pipeline and create a default job called 'main'
+     * @method create
+     * @param  {Object}   config           Config object to create the pipeline with
+     * @param  {String}   config.scmUrl    The scmUrl of the repository
+     * @param  {Function} callback         Callback function
+     */
+    create(config, callback) {
+        const pipelineId = hashr.sha1(config.scmUrl);
+        const getConfig = {
+            table,
+            params: {
+                id: pipelineId
+            }
+        };
+
+        /* eslint-disable consistent-return */
+        this.datastore.get(getConfig, (error, data) => {
+            if (data) {
+                return callback(new Error('scmUrl needs to be unique'));
+            }
+            const createTime = Date.now();
+            const platform = 'generic@1';
+            const pipelineData = hoek.applyToDefaults({
+                createTime,
+                platform,
+                configUrl: config.scmUrl
+            }, config);
+            const pipelineConfig = {
+                table,
+                params: {
+                    id: pipelineId,
+                    data: pipelineData
+                }
+            };
+
+            async.series([
+                async.apply(this.datastore.save, pipelineConfig),
+                (next) => {
```

Comment:
Should the pipeline model handle the creation of a job, or should the `plugin-pipeline` handle the creation of the job?